test/src/com/xilinx/rapidwright/design/TestDesign.java

```diff
@@ -440,4 +443,49 @@ public void testPlaceIOB(String partName, String pkgPin) {
                 : design.getNet(ibufInst.getName() + "/O");
         Assertions.assertEquals(siteInst.getNetFromSiteWire("DOUT"), dout);
     }
+
+    private void ensureDesignInSLR(Design d, int expectedSLR) {
+        for (SiteInst si : d.getSiteInsts()) {
+            Assertions.assertEquals(expectedSLR, si.getTile().getSLR().getId());
+        }
+        for (Net n : d.getNets()) {
+            for (PIP p : n.getPIPs()) {
+                Assertions.assertEquals(expectedSLR, p.getTile().getSLR().getId());
+            }
+        }
+    }
+
+    /**
+     * Tests relocating and retargeting a Picoblaze design from a vu3p to each of
+     * the three SLRs in a vu9p since all of the SLRs between the devices are
+     * relocation compatible.
+     * 
+     * @param tempDir Temp directory to write out results.
+     */
+    @Test
+    public void testRetargetPart(@TempDir Path tempDir) {
+        String targetPartName = "xcvu9p-flgb2104-2-i";
+        Part targetPart = PartNameTools.getPart(targetPartName);
+        Device targetDevice = Device.getDevice(targetPart);
+        for (int slr = 0; slr < targetDevice.getNumOfSLRs(); slr++) {
+            Design d = RapidWrightDCP.loadDCP("picoblaze_ooc_X10Y235.dcp");
+            assert (d.getDevice().getName().equals("xcvu3p"));
+            int tileDX = 0;
+            int tileDY = slr * targetDevice.getMasterSLR().getNumOfClockRegionRows()
+                             * targetPart.getSeries().getCLEHeight();
+            Assertions.assertTrue(d.retargetPart(targetPart, tileDX, tileDY));
+            Path output = tempDir.resolve("retarget_" + slr + ".dcp");
+
+            Assertions.assertEquals(targetPartName, d.getPartName());
+            ensureDesignInSLR(d, slr);
+
+            d.writeCheckpoint(output);
+
+            Design d2 = Design.readCheckpoint(output);
+
+            Assertions.assertEquals(targetPartName, d2.getPartName());
+            ensureDesignInSLR(d2, slr);
```

Comment:
Should we check that Vivado can accept the retargeted/relocated result?

Reply:
Yes, I'll add that.  I'll also try adding a test to see if we can invert the process and arrive back at what we started with as this should be a reversible process.